store/root/store.go

```diff
@@ -174,8 +174,23 @@ func (s *Store) Query(storeKey string, version uint64, key []byte, prove bool) (
 	}
 
 	val, err := s.stateStore.Get(storeKey, version, key)
-	if err != nil {
-		return store.QueryResult{}, err
+	if err != nil || val == nil {
+		// fallback to querying SC backend if not found in SS backend
+		//
+		// Note, this should only used during migration, i.e. while SS and IAVL v2
+		// are being asynchronously synced.
+		if val == nil {
+			bz, scErr := s.stateCommitment.Get(storeKey, version, key)
+			if scErr != nil {
+				return store.QueryResult{}, fmt.Errorf("failed to query SC store: %w", scErr)
+			}
+
+			val = bz
+		}
+
+		if err != nil {
+			return store.QueryResult{}, fmt.Errorf("failed to query SS store: %w", err)
```

Comment:
it looks weird, why error handling here?

Reply:
as opposed to?

Reply:
OK, let's clarify in backward PR, I just wanted to avoid the stateStore access while migrating but not appropriate time